packages-exp/app-check-exp/src/factory.ts

```diff
@@ -29,7 +29,11 @@ import { Provider } from '@firebase/component';
  * AppCheck Service class.
  */
 export class AppCheckService implements AppCheck, _FirebaseService {
-  constructor(public app: FirebaseApp) {}
+  platformLoggerProvider: Provider<'platform-logger'>;
+
+  constructor(public app: FirebaseApp) {
+    this.platformLoggerProvider = _getProvider(app, 'platform-logger');
```

Comment:
Providers should be passed in to the service class constructor in the component factory, so that all dependencies are tracked in the component factory.

Reply:
Done.